### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by benshidanny11**
#### Logic

- Each class from the [UML class diagram](./images/catalog_of_my_things.png) was created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes were implemented.
- All properties visible in the diagram were defined and set up in the constructor method. 
- All methods visible in the diagram were implemented:
    - add_item method (in all classes that have that method)
    - can_be_archived?() in the Item class
    - move_to_archive() in the Item class
    - can_be_archived?() in the Book class
    - can_be_archived?() in the MusicAlbum class
    - can_be_archived?() in the Movie class
    - can_be_archived?() in the Game class
- Added unit tests for all implemented methods.

#### User interface

- Created a main.rb file as app entry-point.
- Our console app, at the start:
    - Presents the user with a list of options to perform.
    - Lets users choose an option.
    - If needed, asks for parameters for the option.
    - Has a way to quit the app.
- The following options are available:
    - List all books
    - List all music albums
    - List all movies
    - List of games
    - List all genres (e.g 'Comedy', 'Thriller')
    - List all labels (e.g. 'Gift', 'New')
    - List all authors (e.g. 'Stephen King')
    - List all sources (e.g. 'From a friend, 'Online shop')
    - Add a book
    - Add a music album
    - Add a movie
    - Add a game
 - All data should be preserved by saving collections in .json files.

### Database
- Created a schema.sql file with tables that will be analogical to the structure of the classes in our app. As we cannot implement inheritance in the database tables - We added all properties and associations from the parent Item class as table columns to all tables based on the child classes.

### Project documentation

We recorded a video presenting the features of the project we built. It is a video with a **maximum length of 5 minutes**. The content of the video includes:

- A description of the project.
- A demo of the project features.

https://user-images.githubusercontent.com/100702612/195339859-e9bd2d12-1c5e-4b3f-b244-8a87c28bbee6.mp4

### CODE REVIEW

File: item.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class Item
+  attr_reader :id
+  attr_accessor :published_date, :archived, :genre, :author, :label
+
+  def initialize(published_date:, archived: false)
+    @id = Random.rand(1..1000)
+    @published_date = published_date
+    @archived = archived
+  end
+
```
Comment (gemmen29) [inline item.rb:12]: - [x] Please add the method that defines the other side of the relation between the `Item` class and `Genre` class like:
```
  def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
  end
```
- [x] Please also add the method that defines the other side of the relation between the `Item` class and the `Author` class.
- [x] Please also add the method that defines the other side of the relation between the `Item` class and the `Label` class.

File: README.md [hunk 2]
```diff
@@ -1 +1,53 @@
-# Ruby-Capstone-Project
\ No newline at end of file
+# Ruby Bank
```
Comment (gemmen29) [inline README.md:1]: - [ ] Kindly stick with the requirement of the video presentation to have a maximum length of 5 minutes as per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#:~:text=It%20is%20a%20video%20with%20a%20maximum%20length%20of%205%20minutes).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: item.rb (hunk 1 above)
Comment (Said-laasri) [reply item.rb:12]: @gemmen29 thanks added methodes

File: README.md (hunk 2 above)
Comment (Said-laasri) [reply README.md:1]: @gemmen29  thanks for review

**Review by DammyShittu — APPROVED** [review]
Hi @benshidanny11 @Said-laasri @rex-9,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**